sdk/keyvault/azure-keyvault-keys/migration_guide.md

```diff
@@ -0,0 +1,271 @@
+
+```python
+keys = client.get_keys(vault_base_url="https://my-key-vault.vault.azure.net/")
+
+for key in keys:
+    print(key.attributes.created)
+```
+
+Now in `azure-keyvault-keys` you can list the properties of keys in a vault with the `list_properties_of_keys` method. This returns an iterator-like object containing `KeyProperties` instances.
+
+```python
+keys = key_client.list_properties_of_keys()
+
+for key in keys:
+    print(key.name)
+    print(key.created_on)
+```
+
+### Delete a key
+
+In `azure-keyvault` you could delete all versions of a key with the `delete_key` method. This returned information about the deleted key (as a `DeletedKeyBundle`), but you could not poll the deletion operation to know when it completed. This would be valuable information if you intended to permanently delete the deleted key with `purge_deleted_key`.
+
+```python
+deleted_key = client.delete_key(vault_base_url="https://my-key-vault.vault.azure.net/", key_name="key-name")
+
+# this purge would fail if deletion hadn't finished
+client.purge_deleted_key(vault_base_url="https://my-key-vault.vault.azure.net/", key_name="key-name")
+```
+
+Now in `azure-keyvault-keys` you can delete a key with `begin_delete_key`, which returns a long operation poller object that can be used to wait/check on the operation. Calling `result()` on the poller will return information about the deleted key (as a `DeletedKey`) without waiting for the operation to complete, but calling `wait()` will wait for the deletion to complete. Again, `purge_deleted_key` will permanently delete your deleted key and make it unrecoverable.
+
+```python
+deleted_key_poller = key_client.begin_delete_key(name="key-name")
+deleted_key = deleted_key_poller.result()
+
+deleted_key_poller.wait()
+key_client.purge_deleted_key(name="key-name")
+```
+
+### Perform cryptographic operations
+
+In `azure-keyvault` you could perform cryptographic operations with keys by using the `encrypt`/`decrypt`, `wrap_key`/`unwrap_key`, and `sign`/`verify` methods. Each of these methods accepted a vault endpoint, key name, key version, and algorithm along with other parameters.
+
+```python
+from azure.keyvault import KeyId
+
+key_bundle = client.create_key(
+    vault_base_url="https://my-key-vault.vault.azure.net/",
+    key_name="key-name",
+    kty="RSA"
+)
+key = key_bundle.key
+key_id = KeyId(key.kid)
+key_version = key_id.version
+
+plaintext = b"plaintext"
+
+# encrypt data using the key
+operation_result = client.encrypt(
+    vault_base_url="https://my-key-vault.vault.azure.net/",
+    key_name="key-name",
+    key_version=key_version,
+    algorithm="RSA-OAEP-256",
+    value=plaintext
+)
+ciphertext = operation_result.result
+```
+
+Now in `azure-keyvault-keys` you can perform these cryptographic operations by using a `CryptographyClient`. The key used to create the client will be used for these operations.
```

Comment:
Another difference is that `CryptographyClient` performs operations locally when it has or can get the key material whereas every `KeyVaultClient` crypto operation is performed by Key Vault.